homeassistant/components/control4/light.py

```diff
@@ -0,0 +1,206 @@
+        update_method=async_update_data_dimmer,
+        update_interval=timedelta(seconds=scan_interval),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await non_dimmer_coordinator.async_refresh()
+    await dimmer_coordinator.async_refresh()
+
+    items_of_category = await get_items_of_category(hass, entry, CONTROL4_CATEGORY)
+    for item in items_of_category:
+        if item["type"] == CONTROL4_ENTITY_TYPE:
+            item_name = item["name"]
+            item_id = item["id"]
+            item_parent_id = item["parentId"]
+            item_is_dimmer = item["capabilities"]["dimmer"]
+
+            if item_is_dimmer:
+                item_coordinator = dimmer_coordinator
+            else:
+                item_coordinator = non_dimmer_coordinator
+
+            for parent_item in items_of_category:
+                if parent_item["id"] == item_parent_id:
+                    item_manufacturer = parent_item["manufacturer"]
+                    item_device_name = parent_item["name"]
+                    item_model = parent_item["model"]
+            async_add_entities(
+                [
+                    Control4Light(
+                        hass,
+                        entry,
+                        item_coordinator,
+                        item_name,
+                        item_id,
+                        item_device_name,
+                        item_manufacturer,
+                        item_model,
+                        item_parent_id,
+                        item_is_dimmer,
+                    )
+                ],
+                True,
+            )
+
+
+class Control4Light(Control4Entity, LightEntity):
+    """Control4 light entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        coordinator: DataUpdateCoordinator,
+        name: str,
+        idx: int,
+        device_name: str,
+        device_manufacturer: str,
+        device_model: str,
+        device_id: int,
+        is_dimmer: bool,
+    ):
+        """Initialize Control4 light entity."""
+        super().__init__(
+            hass,
+            entry,
+            coordinator,
+            name,
+            idx,
+            device_name,
+            device_manufacturer,
+            device_model,
+            device_id,
+        )
+        self._is_dimmer = is_dimmer
+        self._c4_light = None
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._c4_light = C4Light(self.director, self._idx)
+
+    @property
+    def is_on(self):
+        """Return whether this light is on or off."""
+        return bool(self._coordinator.data[self._idx]["value"] > 0)
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        if self._is_dimmer:
+            return self._coordinator.data[self._idx]["value"] * 2.55
```

Comment:
Do we need to round this value to an integer?

Reply:
I'm not sure what you mean - this seems to work fine in my local HA instance as is? Should I be copying this to `int`?

Reply:
The brightness should be an integer between 0-255. Currently we're returning a float. I suggest rounding the value to an integer with `round` and no decimals.
```suggestion
            return round(self._coordinator.data[self._idx]["value"] * 2.55)
```